packages/telemetry/src/index.ts

```diff
@@ -0,0 +1,261 @@
+import type { Span, SpanOptions, Tracer, Context } from '@opentelemetry/api'
+import type { BasicTracerProvider, SimpleSpanProcessor, BatchSpanProcessor, SpanExporter } from '@opentelemetry/sdk-trace-base'
+import type { DetectorSync } from '@opentelemetry/resources'
+
+import openTelemetry/*, { diag, DiagConsoleLogger, DiagLogLevel }*/ from '@opentelemetry/api'
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions'
+import { Resource, detectResourcesSync } from '@opentelemetry/resources'
+
+const types = ['child', 'root'] as const
+
+type AttachType = typeof types[number];
+
+type contextObject = { traceparent?: string }
+
+export type startSpanOptions = {
+  name: string
+  attachType?: AttachType
+  active?: boolean
+  opts?: SpanOptions
+}
+
+// Extend the span type to include span.name
+type NamedSpan = Span & { name: string }
+
+export type findActiveSpanOptions = (element: NamedSpan, index: number) => boolean
```

Comment:
Element can be undefined if the telemetry is a no-op? is this the same for index?
```suggestion
export type findActiveSpanOptions = (element?: NamedSpan, index: number) => boolean
```

Reply:
it wouldn't be undefined, it just wouldn't be called.

Reply:
If thats the case do we need to `?` in https://github.com/cypress-io/cypress/pull/26305/files#diff-f1ed106c3c8949c2deb567f52a44a790333804de9e892ecd261c433b69632a92R10?

Reply:
nope, we sure don't, thanks for making me reconsider that